lab_6_pipeline/pipeline.py

```diff
@@ -62,11 +107,21 @@ def __init__(
             corpus_manager (CorpusManager): CorpusManager instance
             analyzer (LibraryWrapper | None): Analyzer instance
         """
+        self._corpus = corpus_manager
+        self.analyzer = analyzer
 
     def run(self) -> None:
         """
         Perform basic preprocessing and write processed text to files.
         """
+        files = [article.text for article in self._corpus.get_articles().values()]
+        if self.analyzer:
+            files = self.analyzer.analyze(files)
+        for i, article in enumerate(self._corpus.get_articles().values()):
+            to_cleaned(article)
+            if self.analyzer and files:
```

Comment:
Why `and files`?

Reply:
because the method we use requires a string to collect info for conllu and in case we won't get articles we do not need to pass anything because we will have no information for linguistic analysis